odra-casper/wasm-env/src/host_functions.rs

```diff
@@ -574,42 +577,24 @@ fn create_constructor_group(contract_package_hash: ContractPackageHash) -> URef
     .unwrap_or_revert()
 }
 
-fn revoke_access_to_constructor_group(
-    contract_package_hash: ContractPackageHash,
-    constructor_access: URef
-) {
+fn revoke_access_to_constructor_group(package_hash: PackageHash, constructor_access: URef) {
     let mut urefs = BTreeSet::new();
     urefs.insert(constructor_access);
-    storage::remove_contract_user_group_urefs(
-        contract_package_hash,
-        consts::CONSTRUCTOR_GROUP_NAME,
-        urefs
-    )
-    .unwrap_or_revert();
+    storage::remove_contract_user_group_urefs(package_hash, consts::CONSTRUCTOR_GROUP_NAME, urefs)
+        .unwrap_or_revert();
 }
 
 /// Returns address based on a [`CallStackElement`].
 ///
 /// For `Session` and `StoredSession` variants it will return account hash, and for `StoredContract`
 /// case it will use contract hash as the address.
-fn call_stack_element_to_address(call_stack_element: CallStackElement) -> Address {
-    match call_stack_element {
-        CallStackElement::Session { account_hash } => Address::try_from(account_hash)
-            .map_err(|e| ApiError::User(ExecutionError::from(e).code()))
-            .unwrap_or_revert(),
-        CallStackElement::StoredSession { account_hash, .. } => {
-            // Stored session code acts in account's context, so if stored session
-            // wants to interact, caller's address will be used.
-            Address::try_from(account_hash)
-                .map_err(|e| ApiError::User(ExecutionError::from(e).code()))
-                .unwrap_or_revert()
-        }
-        CallStackElement::StoredContract {
-            contract_package_hash,
-            ..
-        } => Address::try_from(contract_package_hash)
-            .map_err(|e| ApiError::User(ExecutionError::from(e).code()))
-            .unwrap_or_revert()
+fn caller_to_address(caller: Caller) -> Address {
```

Comment:
Maybe replace with `From` implemented for `Adderss` directly?

Reply:
Done.